### ISSUE TITLE
I'm confused what the `base` option is supposed to do

### ISSUE THREAD

**Issue by matthew-dean** [peter-evans/create-pull-request#2841]
### Subject of the issue

So, here's what we'd like to do / automate

1. When something is merged to `master`, we'd like to create a PR to merge `master` into `dev`.
2. So, to do that, we need to create a branch from `master` using this action. So the first step of our script looks like this:

```yaml
    steps:
      - name: Checkout repository
        uses: actions/checkout@v4
        with:
          ref: master
```
Next, my assumption was that I need to create the PR using this action, and set the base branch where I will be merging to, so it looks like this:
```yaml
      - name: Create PR to merge master into dev
        id: create_pr
        uses: peter-evans/create-pull-request@v3
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          branch: chore/patch-master-to-dev
          base: dev
          title: Merge master into dev
          body: This PR is automatically created to merge master into dev.
          commit-message: '[automated] Merge master to dev'
```
However, what happens in reality is that this action, instead of creating a Github PR that targets `dev` as the base, it does the following:
```sh
Switched to and reset branch 'f7fcf4ad-2020-4869-b9cb-8949c6662f0b'
  /usr/bin/git -c protocol.version=2 fetch --no-tags --progress --no-recurse-submodules --force origin dev:dev
  /usr/bin/git -c protocol.version=2 fetch --no-tags --progress --no-recurse-submodules origin chore/patch-master-to-dev:refs/remotes/origin/chore/patch-master-to-dev
  fatal: couldn't find remote ref chore/patch-master-to-dev
  Pull request branch 'chore/patch-master-to-dev' does not exist yet.
  /usr/bin/git checkout --progress -B chore/patch-master-to-dev f7fcf4ad-2020-4869-b9cb-8949c6662f0b --
  Switched to a new branch 'chore/patch-master-to-dev'
  /usr/bin/git rev-list --right-only --count dev...chore/patch-master-to-dev
  0
  Branch 'chore/patch-master-to-dev' is not ahead of base 'dev' and will not be created
```
So it appears it first fetches dev (?), and I guess uses that to base the new branch off of? Therefore the PR that is created is no longer ahead of `dev`, therefore the action fails.

But...I can't see any other way / option to set what base branch the PR targets. What am I missing?

In other words, how do you get this action to do this:

<img width="462" alt="Screenshot 2024-04-04 at 10 18 57 AM" src="https://github.com/peter-evans/create-pull-request/assets/414752/4824a63e-72b2-48f3-bd65-d7d81372de44">

... with `chore/patch-master-to-dev` having been branched from `master`? This is especially confusing because Github labels that as `base` which doesn't appear to be what the `base` option of this action does.

**Comment by peter-evans** [comment]
Hi @matthew-dean 

For this use case, please follow this example:
https://github.com/peter-evans/create-pull-request/blob/main/docs/examples.md#keep-a-branch-up-to-date-with-another

Following that example, you will need to be checking out `dev`, updating it from `master`, and then allowing the action to create the PR. You don't need to use the `base` input because the action will already know that `dev` is the base.

If you get stuck let me know.
